File: config/minio.js
const SDK = require('minio');

module.exports.minio = {

    settings: {

      bucket: 'illuminate' // Must be lowercase or else it will error

    },

    Minio: new SDK.Client({

      endPoint: 'play.minio.io',
      port: 9000,
      secure: true,
      accessKey: 'Q3AM3UQ867SPQQA43P2F', 
      secretKey: 'zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG',

  }),
}

File: modules/minio.go
package modules

import (
	"log"

	"github.com/RumbleFrog/Illuminate/helpers"
	"github.com/minio/minio-go"
)

// MinioClient Pointers for Minio connection
var MinioClient *minio.Client

// MinioConnect fufills the MinioClient pointer
func MinioConnect(creds *helpers.MinioCredentials) {
	var err error

	MinioClient, err = minio.New(creds.Endpoint, creds.AccessKey, creds.PrivateKey, creds.Secure)

	if err != nil {
		log.Fatalln(err)
	}

	var exists bool

	exists, err = MinioClient.BucketExists(creds.Bucket)

	if err != nil {
		log.Fatalln(err)
	}

	if !exists {
		err = MinioClient.MakeBucket(creds.Bucket, "us-east-1")

		if err != nil {
			log.Fatalln(err)
		}
	}
}
